File: user_spring/src/main/java/com/kjj/user/entity/Order.java
package com.kjj.user.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Index;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;


@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    private User user;

    private String menu;

    private int cost;

    @Index(name = "order_date_index")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate orderDate;

    private boolean recognize;
    @Column(columnDefinition = "BOOLEAN DEFAULT false")
    private boolean expired;
    @Column(columnDefinition = "BOOLEAN DEFAULT false")
    private boolean payment;

    public void setMenuAndRecognizeTrue(String menuName) {
        this.menu = menuName;
        recognize = true;
    }

    public void setRecognizeToCancel() {
        recognize = false;
    }

    public void setExpiredTrue() {
        expired = true;
    }

    public void setPaymentTrue() {
        payment = true;
    }

    public static Order createNewOrder(User user, String menu, int cost, LocalDate date, boolean type) {
        return new Order(
                null,
                user,
                menu,
                cost,
                date,
                type,
                false,
                false
        );
    }
}

File: user/src/main/kotlin/com/kjj/user/entity/Order.kt
package com.kjj.user.entity

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "orders",
    indexes = arrayOf(
        Index(name = "order_date_index", columnList = "orderDate"))
)
class Order (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,

    @ManyToOne(fetch = FetchType.LAZY)
    val user: User,

    var menu: String,
    val cost: Int,
    val orderDate: LocalDate,
    var recognize: Boolean,
    var expired: Boolean,
    var payment: Boolean
){
    companion object {
        fun createNewOrder(
            user: User,
            menu: String,
            cost: Int,
            date: LocalDate,
            type: Boolean)
        : Order {
            return Order(
                null,
                user,
                menu,
                cost,
                date,
                type,
                false,
                false)
        }
    }

    fun setMenuAndRecognizeTrue(menuName: String) {
        menu = menuName
        recognize = true
    }

    fun setRecognizeToCancel() {
        recognize = false
    }

    fun setExpiredTrue() {
        expired = true
    }

    fun setPaymentTrue() {
        payment = true
    }
}